test/lib.js

```diff
@@ -232,3 +232,25 @@ test('download', async t => {
 
   return pify(fs.unlink)(file);
 });
+
+test('download not an image', async t => {
+  const options = {
+    dir: __dirname
+  };
+  const url = 'https://source.unsplash.com/'
+  const stateStack = [];
+  return download(options, url, state => stateStack.push(state)).catch((err)=>{
```

Comment:
I know you pulled this from my example above (thank you for keeping a similar code style), but I have no idea what `stateStack` does. Do you? I can't remember what it was for.

Reply:
stateStack contains the progress in % but I think it's useless in the test. I'll remove it.